### PR TITLE
List aggregate events performance improvements

### PR DESCRIPTION
**Pull request by stefanvozd**
Improve performance for list aggregate events by reading segments in parallel and prefetching events from the event source

### CODE REVIEW

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/SegmentBasedEventStore.java [hunk 1]
```diff
@@ -147,7 +130,6 @@ public void processEventsPerAggregate(String aggregateId, long firstSequenceNumb
                                                                                       eventConsumer,
                                                                                       Long.MAX_VALUE,
                                                                                       minToken));
-        aggregateReadTimer.record(System.currentTimeMillis() - before, TimeUnit.MILLISECONDS);
```
Comment (MGathier) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/SegmentBasedEventStore.java:150]: I know that customers are monitoring the aggregateReadTimer, if we no longer record the values this must be documented for users.

Comment (stefanvozd) [reply axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/SegmentBasedEventStore.java:150]: aggregateReadTimer is a duplicate of event_stream metrics, and event_stream metric is more granular, it records duration on different layers. I agree that we should document it

**Review by MGathier — APPROVED** [review]

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/SegmentBasedEventStore.java [hunk 2]
```diff
@@ -56,9 +75,9 @@
     protected final StorageProperties storageProperties;
     protected final EventTypeContext type;
     protected final Set<Runnable> closeListeners = new CopyOnWriteArraySet<>();
-    private final Timer aggregateReadTimer;
     private final Timer lastSequenceReadTimer;
     protected final SegmentBasedEventStore next;
+    private final int PREFETCH_SEGMENT_FILES = 2;
```
Comment (m1l4n54v1c) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/SegmentBasedEventStore.java:80]:
```suggestion
    private static final int PREFETCH_SEGMENT_FILES = 2;
```

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/SegmentBasedEventStore.java [hunk 3]
```diff
@@ -74,16 +93,11 @@ public SegmentBasedEventStore(EventTypeContext eventTypeContext, IndexManager in
...
-                                                     Tags.of(MeterFactory.CONTEXT,
-                                                             eventTypeContext.getContext(),
-                                                             "type",
-                                                             eventTypeContext.getEventType().toString()));
         this.lastSequenceReadTimer = meterFactory.timer(BaseMetricName.AXON_LAST_SEQUENCE_READTIME,
-                                                        Tags.of(MeterFactory.CONTEXT,
-                                                                eventTypeContext.getContext(),
-                                                                "type",
-                                                                eventTypeContext.getEventType().toString()));
+                Tags.of(MeterFactory.CONTEXT,
```
Comment (m1l4n54v1c) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/SegmentBasedEventStore.java:97]: Looks like your formatter is not up-to-date. Please check Axon code style.

**Review by m1l4n54v1c — APPROVED** [review]
LGTM.

**Pull request merged**